src/twisted/protocols/_smb/tests.py

```diff
@@ -0,0 +1,363 @@
+#!/usr/bin/python3
+
+import calendar
+import io
+import os
+import re
+import socket
+import struct
+import sys
+import unittest as python_unittest
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Match,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+    get_type_hints,
+)
+
+from zope.interface import implementer
+
+import attr
+
+from twisted.cred import checkers, credentials, portal
+from twisted.internet import reactor
+from twisted.internet.defer import Deferred
+from twisted.internet.interfaces import IProcessTransport, IReactorProcess, IReactorTCP
+from twisted.internet.protocol import ProcessProtocol
+from twisted.logger import Logger, globalLogBeginner, textFileLogObserver
+from twisted.protocols._smb import _base, core, ntlm, security_blob
+from twisted.protocols._smb.ismb import (
+    IFilesystem,
+    IPipe,
+    IPrinter,
+    ISMBServer,
+    NoSuchShare,
+)
+from twisted.protocols._smb.ntlm import NTLMFlag
+from twisted.python.failure import Failure
+from twisted.trial import unittest
+
+log = Logger()
+observers = [textFileLogObserver(sys.stdout)]
+globalLogBeginner.beginLoggingTo(observers)
+
+
+@attr.s
+class FakeStruct:
+    one = _base.short()
+    two = _base.byte()
+    three = _base.single(4.2)
+    four = _base.octets(3)
+    five = _base.long(424242, locked=True)
+
+
+@attr.s
+class FakeStruct2:
+    i = _base.short()
+    b = _base.byte()
+    s = _base.octets(3)
+
+
+class TestBase(unittest.TestCase):
+    def test_base_pack(self) -> None:
```

Comment:
Tests need docstrings so that maintainers understand what properties they're testing. They can (and should!) be very short, per https://jml.io/test-docstrings/ making a short assertion about the specific property they're testing.